ingestion/src/metadata/ingestion/source/pipeline/databrickspipeline/metadata.py

```diff
@@ -0,0 +1,231 @@
+                    id=self.context.pipeline_service.id.__root__, type="pipelineService"
+                ),
+            )
+
+        except TypeError as err:
+            logger.debug(traceback.format_exc())
+            logger.warning(
+                f"Error building Databricks Pipeline information from {pipeline_details}."
+                f" There might be Databricks Jobs API version incompatibilities - {err}"
+            )
+        except ValidationError as err:
+            logger.debug(traceback.format_exc())
+            logger.warning(
+                f"Error building pydantic model for {pipeline_details} - {err}"
+            )
+        except Exception as err:
+            logger.debug(traceback.format_exc())
+            logger.warning(f"Wild error ingesting pipeline {pipeline_details} - {err}")
+
+    def get_tasks(self, pipeline_details: dict) -> List[Task]:
+        task_list = []
+        self.append_context(key="job_id_list", value=pipeline_details["job_id"])
+
+        downstream_tasks = self.get_downstream_tasks(
+            pipeline_details["settings"]["tasks"]
+        )
+        for task in pipeline_details["settings"]["tasks"]:
+            task_list.append(
+                Task(
+                    name=task["task_key"],
+                    displayName=task["task_key"],
+                    taskType=self.get_task_type(task),
+                    downstreamTasks=downstream_tasks.get(task["task_key"], []),
+                )
+            )
+
+        return task_list
+
+    def get_task_type(self, task):
+        task_key = "undefined_task_type"
+        for key in task.keys():
+            if key.endswith("_task"):
+                task_key = key
+
+        return task_key
+
+    def get_downstream_tasks(self, workflow):
+        task_key_list = [task["task_key"] for task in workflow]
+
+        dependent_tasks = self.get_dependent_tasks(workflow)
+
+        downstream_tasks = {}
+
+        for task_key, task_depend_ons in dependent_tasks.items():
+            if task_depend_ons:
+                for task in task_depend_ons:
+                    if task in downstream_tasks:
+                        downstream_tasks[task].append(task_key)
+                    else:
+                        downstream_tasks[task] = [task_key]
+
+        for task in task_key_list:
+            if task not in downstream_tasks:
+                downstream_tasks[task] = []
+
+        return downstream_tasks
+
+    def get_dependent_tasks(self, workflow):
+        dependent_tasks = {}
+
+        for task in workflow:
+            depends_on = task.get("depends_on")
+            if depends_on:
+                dependent_tasks[task["task_key"]] = [v["task_key"] for v in depends_on]
+            else:
+                dependent_tasks[task["task_key"]] = None
+
+        return dependent_tasks
+
+    def yield_pipeline_status(self, pipeline_details) -> Iterable[OMetaPipelineStatus]:
+        for job_id in self.context["job_id_list"]:
```

Comment:
Last comment 🙏 

Is this subscription correct? or should this become `self.context.job_id_list`?

Reply:
Updated. I was using the `self.context` variable incorrectly. Your other comment was really helpful!